src/manifold/src/quickhull.cpp

```diff
@@ -801,11 +816,22 @@ void QuickHull::reclaimToIndexVectorPool(std::unique_ptr<Vec<size_t>>& ptr) {
   indexVectorPool.reclaim(ptr);
 }
 
-bool QuickHull::addPointToFace(typename MeshBuilder::Face& f,
-                               size_t pointIndex) {
+void QuickHull::addPointToFace(typename MeshBuilder::Face& f, size_t pointIndex,
+                               int& pointMutex) {
   const double D =
       getSignedDistanceToPlane(originalVertexData[pointIndex], f.P);
   if (D > 0 && D * D > epsilonSquared * f.P.sqrNLength) {
+    // Only adds point to 1 face
+
+    // Use when ExecutionPolicy::Par
+    // if (AtomicAdd(pointMutex,1)!=0)
+    // return true;
+
+    // For ExecutionPolicy::Seq
+    pointMutex = 1;
+
+    // Ensures atomic addition of point to face
+    f.faceMutex->lock();
     if (!f.pointsOnPositiveSide) {
       f.pointsOnPositiveSide = getIndexVectorFromPool();
```

Comment:
this call is not thread-safe.

Reply:
Lock call isn't thread-safe?

Reply:
I mean the `getIndexVectorFromPool`.

Reply:
Oh right, since the Pool is same for all faces, I should use a lock specifically for pool. Thanks for pointing that out!, that helps a lot.

Reply:
Do note that for every lock you are adding now, you will likely be thinking about how to remove them later...

Reply:
Anyway, for now, the important thing is to make sure it works. We can gradually improve on performance.

Reply:
Yes, I will keep that in mind. Also, the function call happens once per face. So, should not add much overhead.